### PR TITLE
VStream: Change notification stream for vitess events

### PR DESCRIPTION
**Pull request by sougou**
This feature is part of the VReplication work. This change exposes a new `VStream` functionality through VTGate that allows you to stream binlog events from the underlying keyspaces and shards. There are a few things that make it different from streaming directly from the mysql instances:
* You can combine multiple shards or multiple keyspaces into one stream.
* There is a 'GTID' that represents the consolidated position. The format is `ks1:0@MySQL56/<mysql_pos>|ks2:-80@MySQL56/<mysql_pos>`.
* If a replica goes down, it should find another one and continue.

The feature is currently exposed only as a GRPC endpoint. You can see an example of it at `go/vt/vtgate/endtoend/vstream_test.go`.

### CODE REVIEW

**Comment by derekperkins** [comment]
Can you restart a stream from a given "VGTID" to replay events, a la Kafka?

**Comment by dweitzman** [comment]
Hypothetically if someone was using table acls to limit read access, it seems like this might allow bypassing those acls?

(At some point in the future I imagine a world where vitess supports column-level ACLs so people can block access to a PII column in both queries and the update stream)

**Comment by sougou** [comment]
Yeah. This is on my TODO list. I thought I added this to #4587, but it's not there. But I have it on my internal notes. I should compare and add missing things to that issue.

File: go/vt/vttest/local_cluster.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ type LocalCluster struct {
 // This connection should be used for debug/introspection purposes; normal
 // cluster access should be performed through the vtgate port.
 func (db *LocalCluster) MySQLConnParams() mysql.ConnParams {
-	return db.mysql.Params(db.DbName())
+	return db.mysql.Params("")
```
Comment (deepthi) [inline go/vt/vttest/local_cluster.go]: Why this change? This might be causing some of the test failures.

Comment (sougou) [reply go/vt/vttest/local_cluster.go]: I see now. Pushing a more correct fix to LocalCluster:

Comment (sougou) [reply go/vt/vttest/local_cluster.go]:
```

Comment (sougou) [reply go/vt/vttest/local_cluster.go]:
```
The DbName function for the local cluster was returning an
incorrect name previously. It was the keyspace name, which would
not match the db names in mysql. The previous fix stopped using
DbName for setting the mysql parameters, which broke some tests.

The new behavior fixes DbName to return a value only if it was
in OnlyMySQL mode.
```

**Pull request merged**